### PR TITLE
Cockpit 240

### PR DESCRIPTION
**Pull request by mvollmer**
https://mvollmer.github.io/cockpit-project.github.io/blog/cockpit-240.html

### CODE REVIEW

File: _posts/2021-03-17-cockpit-240.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+Cockpit is now also available in Norwegian Bokmål localization. Thank
+you Stein-Aksel Basma for this contribution!  Cockpit is being
+translated in [Weblate](https://translate.fedoraproject.org/projects/cockpit/master/)
+where anyone can help to improve Cockpit.
+
+## Performance metrics: Journal integration
+
+For every event on metrics page relevant journal logs are provided to
+help identify why some event has happened.
```
Comment (allisonkarlitskaya) [inline _posts/2021-03-17-cockpit-240.md:25]:
```suggestion
help identify why the event has happened.
```

File: _posts/2021-03-17-cockpit-240.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+![Journal entries for a spike](/images/metrics-journal.png)
```
Comment (allisonkarlitskaya) [inline _posts/2021-03-17-cockpit-240.md:27]: Without looking at the image, it's not at all clear what "for a spike" means.

File: _posts/2021-03-17-cockpit-240.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+## Machines: support authentication against cloud images
+
+Distro cloud images have locked login accounts by default. The user
+can now specify the root and user account login credentials when
+creating VMs based on cloud images. The cloud image itself will be
+used as the backing store for the VM's disk.
```
Comment (allisonkarlitskaya) [inline _posts/2021-03-17-cockpit-240.md:34]: The last sentence implies that this feature is much larger than the summary implies.

Which way is it?  We only added authentication for existing cloud-image-as-backing-store VMs, or we added the feature entirely?

File: _posts/2021-03-17-cockpit-240.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+![Cloud image account initialization](/images/cloud-init.png)
+
+### Try it out
+
+Cockpit 240 is available now:
+
+ * [For your Linux system](https://cockpit-project.org/running.html)
+ * [Cockpit Source Tarball](https://github.com/cockpit-project/cockpit/releases/tag/230)
+ * [Cockpit Fedora 33](https://bodhi.fedoraproject.org/updates/FEDORA-2021-XXX)
```
Comment (allisonkarlitskaya) [inline _posts/2021-03-17-cockpit-240.md:44]: Needs real urls, of course.

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]

File: _posts/2021-03-17-cockpit-240.md (hunk 3 above)
Comment (KKoukiou) [reply _posts/2021-03-17-cockpit-240.md:34]: Why does it imply that the feature is larger :)? It totally is but I just want to see how you figured it out.

Cloud images can be customized through cloud init in much more deep way than just login credentials (ssh keys, network configuration, adding packages is some of them)

We don't plan to add support for none of these, we will just keep it simple and limited to this feature that I describe here.

File: _posts/2021-03-17-cockpit-240.md (hunk 2 above)
Comment (mvollmer) [reply _posts/2021-03-17-cockpit-240.md:27]: Yes, should be "for an event", to refer back to the text.

File: _posts/2021-03-17-cockpit-240.md (hunk 3 above)
Comment (mvollmer) [reply _posts/2021-03-17-cockpit-240.md:34]: I think we can remove the last sentence. It's not relevant to know what a backing store is to use the new feature.

File: _posts/2021-03-17-cockpit-240.md (hunk 4 above)
Comment (mvollmer) [reply _posts/2021-03-17-cockpit-240.md]: Added.

File: _posts/2021-03-17-cockpit-240.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+For every event on metrics page relevant journal logs are provided to
+help identify why the event has happened.
+
+![Journal entries for an event](/images/metrics-journal.png)
+
+## Machines: support authentication against cloud images
+
+Distro cloud images have locked login accounts by default. The user
+can now specify the root and user account login credentials when
+creating VMs based on cloud images.
```
Comment (garrett) [inline _posts/2021-03-17-cockpit-240.md:33]: First pass:

```suggestion
Linux distribution cloud images use locked login accounts by default. Cockpit supports creating virtual machines based on cloud images using these credentials.
```

Second pass:

```suggestion
Cockpit can now create virtual machines based on cloud images using login credentials.
```

Or, perhaps:

```suggestion
Cockpit now supports login credentials when creating virtual machines from cloud images.
```

Or if "distro" matters:

```suggestion
Cockpit now supports login credentials when creating virtual machines from Linux distribution cloud images.
```

**Review by garrett — CHANGES_REQUESTED** [review]
Lots of suggestions. Some of them are choose-your-own suggestion even. :wink:

File: _posts/2021-03-17-cockpit-240.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+![Cloud image account initialization](/images/cloud-init.png)
+
+### Try it out
+
+Cockpit 240 is available now:
+
+ * [For your Linux system](https://cockpit-project.org/running.html)
+ * [Cockpit Source Tarball](https://github.com/cockpit-project/cockpit/releases/tag/230)
```
Comment (martinpitt) [inline _posts/2021-03-17-cockpit-240.md:42]: 240 :)

**Comment by garrett** [comment]
The cloud init screenshot was just slightly over the max width of images for the blog, so I shaved off a few pixels and optimized the image for file size. I didn't add a border as it's the full page and the background is shaded anyway.

File: _posts/2021-03-17-cockpit-240.md (hunk 3 above)
Comment (martinpitt) [reply _posts/2021-03-17-cockpit-240.md]: As someone who knows cloud images, I find the description a bit confusing. Maybe s/locked/no/ in the first sentence, dropping the redundant "when creating VMs based on cloud images", and pointing out the vehicle:

| The user can now specify the root and user account login credentials, which will be passed to [cloud-init](https://cloud-init.io/).

This is concise, not too technical, and hints at what's going on for people who actually use cloud images.

File: _posts/2021-03-17-cockpit-240.md (hunk 6 above)
Comment (mvollmer) [reply _posts/2021-03-17-cockpit-240.md:42]: Damn.

File: _posts/2021-03-17-cockpit-240.md (hunk 3 above)
Comment (mvollmer) [reply _posts/2021-03-17-cockpit-240.md]: Done.

File: _posts/2021-03-17-cockpit-240.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.
+
+Here are the release notes from Cockpit version 240.
+
+## New localization: Norwegian Bokmål
+
+Cockpit is now also available in Norwegian Bokmål localization. Thank
+you Stein-Aksel Basma for this contribution!  Cockpit is being
+translated in [Weblate](https://translate.fedoraproject.org/projects/cockpit/master/)
```
[the remark this answers is not in the document]
Comment (mvollmer) [reply _posts/2021-03-17-cockpit-240.md:19]: Done.

File: _posts/2021-03-17-cockpit-240.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+where anyone can help to improve Cockpit.
+
+## Performance metrics: Journal integration
+
+For every event on metrics page relevant journal logs are provided to
+help identify why the event has happened.
```
[the remark this answers is not in the document]
Comment (mvollmer) [reply _posts/2021-03-17-cockpit-240.md:25]: I'll take the second suggestion.

File: _posts/2021-03-17-cockpit-240.md (hunk 5 above)
Comment (mvollmer) [reply _posts/2021-03-17-cockpit-240.md:33]: I have used @martinpitt's version.

**Review by martinpitt — APPROVED** [review]
Thank you!

**Comment by garrett** [comment]
Hooray! Now it's time to tweet about it! 

:bird: :speech_balloon:

**Comment by garrett** [comment]
I'm thinking of tweeting:

---

```
Just released: Cockpit 240!

https://cockpit-project.org/blog/cockpit-240.html

Includes:
- New localization: Norwegian Bokmål
- Performance metrics: Journal integration
- Machines: Cloud image authentication

And, of course: the usual "bug fixes and enhancements"! 😉
```

**Comment by garrett** [comment]
Might as well have that little joke (what apps, especially Google ones, say generically on the Google Play Store) on the first tweet about updates, right?

**Comment by allisonkarlitskaya** [comment]
"various bug fixes and performance improvements" prior art: https://github.com/cockpit-project/cockpit-project.github.io/blob/main/_posts/2020-11-25-cockpit-233.md :rofl:

**Pull request merged**